util/matchers.go

```diff
@@ -0,0 +1,17 @@
+package util
+
+import (
+	"github.com/prometheus/prometheus/storage/metric"
+)
+
+// SplitFiltersAndMatchers splits empty matchers off, which are treated as filters, see #220
+func SplitFiltersAndMatchers(allMatchers []*metric.LabelMatcher) (filters, matchers []*metric.LabelMatcher) {
+	for _, matcher := range allMatchers {
+		if matcher.Match("") {
```

Comment:
The implementation of that function was a bit suspect, so I thought best to skip it...